tools/reviewer/web/src/app/dashboard/reviews-panel/diff/diff.component.ts

```diff
@@ -17,42 +29,69 @@ import { files } from './mock-files';
 })
 export class DiffComponent implements OnInit, OnDestroy {
   isLoading: boolean = true;
-  filename: string;
-  files: string[] = files;
+  textDiffResponse: TextDiffResponse;
   changes: number[];
-  snapshot: number;
   localThreads: Thread[];
   diff: Diff;
   newCommentSubscription: Subscription;
+  filename: string;
 
   constructor(
     private activatedRoute: ActivatedRoute,
     private differenceService: DifferenceService,
     private firebaseService: FirebaseService,
-    private diffService: DiffService
+    private diffService: DiffService,
+    private encodingService: EncodingService,
+    private notificationService: NotificationService,
+    private http: Http,
   ) {
-    this.changes = this.differenceService.compare(files[0], files[1]);
-
     this.newCommentSubscription = this.diffService.newComment.subscribe(
       param => {
         this.addComment(param.lineNumber, param.comments);
       }
     );
   }
 
-  addComment(lineNumber: number, comments: Comment[]): void {
-    if (comments.length === 1) {
-      // Create a thread, if it's first comment
-      const newThread = new Thread();
-      newThread.setSnapshot(this.snapshot);
-      newThread.setLineNumber(lineNumber);
-      newThread.setFilename(this.filename);
-      newThread.setIsDone(false);
-      newThread.setCommentsList(comments);
-      this.diff.addThreads(newThread);
-    }
+  getFileChanges(): void {
+    const textDiffRequest = new TextDiffRequest();
 
-    this.firebaseService.updateDiff(this.diff).subscribe();
+    // Temperoally hardcoded data
+    // TODO: take the data from somewhere else (url?)
```

Comment:
Is this about how to get the data from a different part of the app?
Maybe @MateMahinKhan has some ideas.

Reply:
No, it's not about it.
It's part of communication with local server, which I don't understand.
In this part of the app we have
1) Diff from firebase
2) filename
How we're going to know, which `commit_id` should be used?

Reply:
Maybe we should put the commit ids to url?

Reply:
So the commit_id should be the last one on the right.
On the left, let's put the first commit for now. Later we will put the last-reviewed one, but we need to mark it as such, which we don't currently.